core/src/main/java/org/kohsuke/stapler/StaplerRequest.java

```diff
@@ -515,6 +517,36 @@ public interface StaplerRequest extends HttpServletRequest {
      * a proxy on the client side.
      *
      * Short cut for {@code getBoundObjectTable().bind(toBeExported).getProxyScript()}
+     *
+     * @deprecated Use {@link #createJavaScriptProxyParameters(Object)} and invoke {@code makeStaplerProxy} yourself.
      */
+    @Deprecated(since = "TODO")
```

Comment:
Nobody is going to care about the version; keep it simple.

```suggestion
    @Deprecated
```